app/javascript/packs/find-out.js

```diff
@@ -0,0 +1,161 @@
+$(() => {
+  const allTaggings = {
+    content_type: 'everything',
+    topic_area: 'all topic areas'
+  }
+  fetchFilteredTaggings(allTaggings)
+  loadDropDownTags()
+})
+
+function loadDropDownTags() {
+  $.get({
+    url: '/tags.json',
+    dataType: 'json',
+  }).done((response) => {
+    const contentTypes = response.filter(tag => tag.data.attributes.tag_type === 'content_type')
+    const contentTypesSelectOptions = contentTypes.map(tag => `<option data-id=${tag.data.attributes.id} value=${tag.data.attributes.title} class="find-out__tag-title">${tag.data.attributes.title}</option>`).join('')
+    const contentTypesDropdown = (`<select><option value='everything' selected>everything</option>${contentTypesSelectOptions}</select`)
+    document.getElementById('find-out__dropdown-content-type').innerHTML = contentTypesDropdown
+
+    const topicAreas = response.filter(tag => tag.data.attributes.tag_type === 'topic_area')
+    const topicAreasSelectOptions = topicAreas.map(tag => `<option data-id=${tag.data.attributes.id} value=${tag.data.attributes.title} class="find-out__tag-title">${tag.data.attributes.title}</option>`).join('')
+    const topicAreasDropdown = (`<select><option value='all topic areas' selected>all topic areas</option>${topicAreasSelectOptions}</select`)
+    document.getElementById('find-out__dropdown-topic-area').innerHTML = topicAreasDropdown
+
+    onTagClick()
+  })
+}
+
+const closeOverlay = () => {
+  $('div.overlay').on('click', (event) => {
+    event.preventDefault()
+    $('div.overlay').html('').removeClass('overlay')
+  })
+}
+
+const openOverlay = () => {
+  $('div.overlay').html('')
+  $('body').append('<div class="overlay"></div>')
+}
+
+const onTagClick = () => {
+  $('select').on('change', (event) => {
+    event.preventDefault()
+
+    const selectedOptions = Array.from(document.getElementsByTagName('option')).filter(tag => tag.selected)
+    const dataObject = {
+      content_type: selectedOptions[0].innerText,
+      topic_area: selectedOptions[1].innerText
+    }
+
+    openOverlay()
+    fetchFilteredTaggings(dataObject)
+  })
+}
+
+const loadTopicAreaNarrative = (title, body) => {
+  $('.find-out__topic-area-narrative-title').text(title)
+  $('.find-out__topic-area-narrative-body').text(body)
+}
+
+const fetchFilteredTaggings = (dataObject) => {
+  $.get({
+    url: '/taggings.json',
+    dataType: 'json',
+    data: dataObject,
+  }).done(response => {
+    loadTopicAreaNarrative(dataObject.topic_area, response.topic_area_narrative)
+
+    const resultsDiv = $('.find-out__results')
+    resultsDiv.empty()
+
+    response.taggings.forEach(res => {
+      if (res.data.attributes.report_id) {
+        createReport(res, resultsDiv)
+      }
+      if (res.data.attributes.event_id) {
+        createEvent(res, resultsDiv)
+      }
+    })
+    closeOverlay()
+  })
+}
+
+// Report api, class and find-out card html
+function createReport(obj, resultsDiv) {
```

Comment:
Instead of using `obj` as a parameter name I usually prefer to use a noun that is a bit more descriptive if I can. What about `tagging` or something similar?

Reply:
Done